src/ORM/DataList.php

```diff
@@ -1367,6 +1420,46 @@ private function addEagerLoadedDataToParent(
         }
     }
 
+    private function fillEmptyEagerLoadedRelations(
+        Query|array $parents,
+        array $parentIDs,
+        string $relationChain,
+        string $relationName,
+        string $relationType,
+        string $relationDataClass,
+        ?string $relationListClass = null,
+        ?array $component = null
+    ): void {
+        foreach ($parentIDs as $id) {
+            // Build the empty list or record
+            switch ($relationType) {
+                case 'has_one':
+                    $dummyData = Injector::inst()->create($relationDataClass);
+                    break;
```

Comment:
`has_one` isn't used currently, but it's there for the sake of completion. It _is_ technically valid to pass in here, and shouldn't throw an exception if we do for some reason pass it in.